worker.go

```diff
@@ -2,155 +2,188 @@ package main
+	if err != nil {
+		return "", fmt.Errorf("discoverCertificateAndUpdateIngress failed to find a certificate for %q: %v",
+			ingress.CertHostname(), err)
 	}
-	return missingARNs, existingARNs
+	ingress.SetCertificateARN(certificateSummary.ID())
+	return certificateSummary.ID(), nil
 }
 
-func createMissingLoadBalancer(awsAdapter *aws.Adapter, certificateARN string) (*aws.LoadBalancer, error) {
-	log.Printf("creating ALB for ARN %q\n", certificateARN)
+func createStack(awsAdapter *aws.Adapter, item *managedItem) {
+	certificateARN := item.ingress.CertificateARN()
+	log.Printf("creating stack for certificate %q / ingress %q", certificateARN, item.ingress)
 
-	lb, err := awsAdapter.CreateLoadBalancer(certificateARN)
+	stackId, err := awsAdapter.CreateStack(certificateARN)
 	if err != nil {
-		return nil, fmt.Errorf("failed to create ALB for certificate %q. %v\n", certificateARN, err)
+		if isAlreadyExistsError(err) {
+			item.stack, err = awsAdapter.GetStack(stackId)
+			if err == nil {
+				return
+			}
+		}
+		log.Printf("createStack(%q) failed: %v", certificateARN, err)
+	} else {
+		log.Printf("stack %q for certificate %q created", stackId, certificateARN)
 	}
-	return lb, nil
 }
 
-func updateIngresses(kubeAdapter *kubernetes.Adapter, ingresses []*kubernetes.Ingress, dnsName string) {
-	for _, ingress := range ingresses {
-		if err := kubeAdapter.UpdateIngressLoadBalancer(ingress, dnsName); err != nil {
-			log.Println(err)
-		} else {
-			log.Printf("updated ingress %v with DNS name %q\n", ingress, dnsName)
-		}
+func isAlreadyExistsError(err error) bool {
+	if awsErr, ok := err.(awserr.Error); ok {
+		return awsErr.Code() == cloudformation.ErrCodeAlreadyExistsException
 	}
+	return false
 }
 
-func deleteOrphanedLoadBalancers(awsAdapter *aws.Adapter, ingresses []*kubernetes.Ingress) error {
-	lbs, err := awsAdapter.FindManagedLoadBalancers()
-	if err != nil {
-		return err
+func updateIngress(kubeAdapter *kubernetes.Adapter, item *managedItem) {
+	if item.stack == nil {
+		return
 	}
-
-	certificateMap := make(map[string]bool)
-	for _, ingress := range ingresses {
-		certificateMap[ingress.CertificateARN()] = true
+	dnsName := strings.ToLower(item.stack.DNSName()) // lower case to satisfy Kubernetes reqs
+	if err := kubeAdapter.UpdateIngressLoadBalancer(item.ingress, dnsName); err != nil {
+		if err != kubernetes.ErrUpdateNotNeeded {
+			log.Println(err)
+		}
+	} else {
+		log.Printf("updated ingress %v with DNS name %q", item.ingress, dnsName)
 	}
+}
 
-	for _, lb := range lbs {
-		if _, has := certificateMap[lb.CertificateARN()]; !has {
-			if err := awsAdapter.DeleteLoadBalancer(lb); err == nil {
-				log.Printf("deleted orphaned load balancer ARN %q\n", lb.ARN())
-			} else {
-				log.Printf("failed to delete orphaned load balancer ARN %q: %v\n", lb.ARN(), err)
-			}
-		}
+func deleteStack(awsAdapter *aws.Adapter, item *managedItem) {
```

Comment:
same as above, errors are swallowed. IF this is intended we should at least make a comment to make this clear.